### PR TITLE
Trip read func

### PR DESCRIPTION
**Pull request by vTrip**
Modified the code to properly read and present trips. Also added in
buttons for ‘edit’ and ‘delete’ minus the functionality.

### CODE REVIEW

File: www/js/controllers.js [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+  TripService.getTrips()
+    .then(function(results) {
+      if (results.length > 0) {
+        $scope.trips = results;
+      } else {
+        // empty state here ?
+      }
+    }, function(results) {
+      // something went wrong here
+      return $q.reject(results.data);
```
Comment (jsuresh) [inline www/js/controllers.js]: In the controller, you'd want to show an error here.

File: www/js/controllers.js [hunk 2]
```diff
@@ -1,7 +1,21 @@
 angular.module('awayapp')
 
 .controller('TripsCtrl', function($scope, TripService) {
-  $scope.trips = TripService.getTrips();
+  TripService.getTrips()
```
Comment (jsuresh) [inline www/js/controllers.js]: Loading placeholder? My usual pattern for this is to abstract this into it's own helper

$scope.trips = withLoading(TripService.getTrips.then(function(res) { .. return results });

```
function withLoading(promise) {
   var result = {
     $loading: true,
     $error: undefined,
     value: undefined
   };

   promise.then(function(val) {
      result.value = val;
      result.$loading = false;
   }).catch(function(err) {
      result.$error = err; /* err.message if you expect error messages */
   });

  return result;
}
```

File: www/js/services.js [hunk 3]
```diff
@@ -4,10 +4,13 @@ myModules.factory("TripService", function($http, $q) {
   return {
     getTrips: function() {
       return $q.when([
-        {name: 'Honeymoon'},
-        {name: 'Some Shit'},
+        {id: '0', name: 'Honeymoon'},
+        {id: '1', name: 'Some Shit'},
```
Comment (jsuresh) [inline www/js/services.js]: haha, I'd leave swear words out of the repo, just in case.

**Comment by jsuresh** [comment]
Bunch of comments, please address and re-upload a PR

**Comment by vTrip** [comment]
Going through them all now - all being noted.

Cheers,

-V

--
Vince Tripodi
Software Development, Praxis
Level 26 / 1 Bligh St, Sydney, NSW 2000
t: 02 8226 8679
m: 0405 186 627
www.hellopraxis.com <http://www.hellopraxis.com/>

**Pull request merged**